lib/spec_rewrite_copy.ts

```diff
@@ -0,0 +1,348 @@
+import * as Vega from "vega"
+import { Transforms, AggregateTransform, None } from "vega"
+import { aggregate, extent, bin } from "vega-transforms";
+import { parse } from "vega-expression"
+//import expr2sql from "./expr2sql"
+import { error, hasOwnProperty } from 'vega-util';
+import { strict } from "assert";
+
+
+
+function percentileContSql(field: string, fraction: number, db: string) {
+  // creates a percentile predicate for a SQL query
+  switch (db.toLowerCase()) {
+    case "postgres":
+      return `PERCENTILE_CONT(${fraction}) WITHIN GROUP (ORDER BY ${field})`;
+    case "duckdb":
+      return `QUANTILE(${field}, ${fraction})`;
+    default:
+      throw Error(`Unsupported database: ${db}`);
+  }
+}
+
+function aggregateOpToSql(op: string, field: string, db: string) {
+  // Converts supported Vega operations to SQL
+  // for the given field.
+  // FixMe: we will need to eventually support the case where
+  // the 'field' is actually a vega expression, which would
+  // require translating vega expressions into SQL.
+  // FixMe: decide what to do for argmax, argmin, and confidence intervals (ci0, ci1).
+  switch (op.toLowerCase()) {
+    case "average":
+      return `AVG(${field})`;
+    case "count":
+    case "valid":
+      return `COUNT(*)`;
```

Comment:
Why don't you write the valid like the missing below? It would mean that the logic for valid is not distributed across here and the where clause.